aspnet/publishing/iis.rst

```diff
@@ -1,116 +1,178 @@
 
-1. Enable the **Web Server (IIS)** server role. In client operating systems (Windows 7 through Windows 10) select **Control Panel > Programs > Programs and Features > Turn Windows features on or off**, and then select **Internet Information Services**.
+.. warning:: 
+  .NET Core applications are hosted via a reverse-proxy between IIS and the Kestrel server. In order to create the reverse-proxy, the `web.config` file must be present at the content root path (i.e., the application base path) of the deployed application, which is the website physical path provided to IIS.
 
-.. image:: pubIIS/_static/rs.png
+  On the application's physical path, there are sensitive files that one would never want to accidently serve, such as **my_application.runtimeconfig.json**, **my_application.xml** (XML Documentation comments), and **my_application.deps.json**. Without the `web.config` file present to create the reverse proxy to Kestrel, IIS would serve these sensitive files openly on the Internet. **Therefore, it is important that the web.config file is never accidently renamed or removed from the deployment.**
 
-2. On the **Role Services** step, remove any items you don't need. The defaults are shown below.
+  In order to mitigate the risk of serving sensitive application files if `web.config` were ever accidently renamed or removed, one may add sensitive files to the **Hidden Segments** of **Request Filtering** or add blanket file extension rules to the **File Name Extensions** of **Request Filtering**. See `Hidden Segments \<hidden Segments\> <https://www.iis.net/configreference/system.webserver/security/requestfiltering/hiddensegments>`__ and `File Name Extensions \<fileExtensions\> <https://www.iis.net/configreference/system.webserver/security/requestfiltering/fileextensions>`__ for more information.
 
-.. image:: pubIIS/_static/role-services.png
+  Keep in mind that if you use the **File Name Extensions** approach that your application will not receive requests nor be able to serve any file with that extension, even static files in your **webroot** that you wish to serve (e.g., JSON or XML files you have placed in `wwwroot` for legitimate file serving). Therefore, specifically naming sensitive files using the **Hidden Segments** approach might be preferable in most scenarios [i.e., explicitly naming **my_application.runtimeconfig.json**, **my_application.xml** (if any XML Documentation comments files are present), **my_application.deps.json**, and any other files that are not explicitly excluded by IIS for static file serving].
```

Comment:
replace all e.g. with "for example"
replace all paths/files with *italic* and remove `code` backticks. I won't make that comment again.